openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -1001,12 +1002,16 @@ def get_bundle_links(library_key):
             opaque_key = libraries_linked[link_data.bundle_uuid].library_key
         except KeyError:
             opaque_key = None
-        # Append the link information:
+        try:
+            latest_version = blockstore_cache.get_bundle_version_number(link_data.bundle_uuid)
+        except BundleNotFound:
+            # if the bundle doesn't exist, it's a broken link, so ignore it
+            continue
```

Comment:
Instead of `continue`, shouldn't we explicitly set `latest_version = None`? Otherwise it looks like `latest_version` will be undefined.

Reply:
@bradenmacdonald this is in a loop, so it will skip appending to `results`

Reply:
Oh I see. However, perhaps we still want to return a result? If I remember correctly, the link still exists and still points to the data in the deleted bundle; the only thing that doesn't exist is the "latest version" of the bundle, since the bundle was deleted. If so, it probably makes sense to still return `LibraryBundleLink` with all the same information, except set `latest_version=None` to indicate that the bundle has been deleted and the link can't be updated (but should continue to work, pointing at the current version/content).

Not returning the link at all would be inconsistent, if the link is actually in use.

Reply:
I agree with @bradenmacdonald 's recommendation, that way it would be more consistent, but I guess in that case you need to refactor the `LibraryBundleLinkSerializer` to also accept nulls in the `latest_version` field

Reply:
@bradenmacdonald @eLRuLL if we do that, will there be issues getting it merged and deployed, since this will be a (minor) breaking change?

Reply:
Updated to use latest_version = 0 if not found, to be consistent with [get_bundle_version_number](https://github.com/open-craft/edx-platform/blob/0f064227f88f23831b5eeb3e8b2dc90aa9d131bb/openedx/core/djangolib/blockstore_cache.py#L137).  What do you think?

Reply:
Good question, but no, we're the only API consumer using links so it shouldn't be a problem.

> Updated to use latest_version = 0 if not found, to be consistent with get_bundle_version_number. What do you think?

Sounds good!